llrt_modules/src/modules/child_process.rs

```diff
@@ -79,30 +79,44 @@ generate_signal_from_str_fn!(
     libc::SIGTERM
 );
 
+#[allow(unused_variables)]
 fn prepare_shell_args(
```

Comment:
The code in this function is much harder to read I feel...
And I am not sure it is more efficient with the push in the string since it might alloc twice vs a concat that will alloc once

Reply:
It _might_ only allocate once but the second push will have capacity since cap is doubled if first needs to allocate. However we can reserve upfront since we’ll know that we’ll push the arguments plus I can push_str them rather than join + push 👌